src/reducers/entities/users.js

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import {combineReducers} from 'redux';
+import {UsersTypes} from 'constants';
+import {profilesToSet, addProfileToSet, removeProfileFromSet} from './helpers';
+
+function currentId(state = '', action) {
+    switch (action.type) {
+    case UsersTypes.RECEIVED_ME:
+        return action.data.id;
+
+    case UsersTypes.LOGOUT_SUCCESS:
+        return '';
+
+    }
+
+    return state;
+}
+
+function myPreferences(state = {}, action) {
+    const nextState = {...state};
+    switch (action.type) {
+    case UsersTypes.RECEIVED_PREFERENCES:
+        const preferences = action.data;
+        for (const p of preferences) {
+            nextState[`${p.category}--${p.name}`] = p.value;
+        }
+        return nextState;
+
+    case UsersTypes.RECEIVED_PREFERENCE:
+        const p = action.data;
+        nextState[`${p.category}--${p.name}`] = p.value;
+        return nextState;
+
+    case UsersTypes.LOGOUT_SUCCESS:
+        return {};
+
+    default:
+        return state;
+    }
+}
+
+function mySessions(state = [], action) {
+    switch (action.type) {
+    case UsersTypes.RECEIVED_SESSIONS:
+        return [...action.data];
+
+    case UsersTypes.RECEIVED_REVOKED_SESSION:
+        let index = -1;
+        const length = state.length;
+        for (let i = 0; i < length; i++) {
+            if (state[i].id === action.data.id) {
+                index = i;
+                break;
+            }
+        }
+        if (index > -1) {
+            return state.slice(0, index).concat(state.slice(index + 1));
+        }
+
+        return [...action.data];
+
+    case UsersTypes.LOGOUT_SUCCESS:
+        return [];
+
+    default:
+        return state;
+    }
+}
+
+function myAudits(state = [], action) {
+    switch (action.type) {
+    case UsersTypes.RECEIVED_AUDITS:
+        return [...action.data];
+
+    case UsersTypes.LOGOUT_SUCCESS:
+        return [];
+
+    default:
+        return state;
+    }
+}
+
+function profiles(state = {items: {}, offset: 0, count: 0}, action) {
```

Comment:
Maybe we could make this initial state into a helper function? We're recreating the same object all over the place